### ISSUE TITLE
Bug(require-schema-suffix): false positive if last method in chain is not `.parse()`

### ISSUE THREAD

**Issue by MartinCura** [marcalexiei/eslint-plugin-zod#71]
### 🧩 Description

Rule not violated if `.parse()` is called immediately but _is_ violated if something else is done in the same line.

### 🪜 How to reproduce

```ts
const somethingAlreadyParsed = z.array(z.string()).parse([]).filter(() => true)
   // ^ Variable declaring Zod schema must have a Schema 
```

No problem:
```ts
const somethingAlreadyParsed = z.array(z.string()).parse([])
```

### ✅ Expected behavior

If i'm not storing the schema into a variable but using it immediately, rule shouldn't be violated even if i do more operations on the parsed/decoded value.

### 💻 System info

```shell
System:
    OS: Linux 6.16 Fedora Linux 42 (Workstation Edition)
    CPU: (22) x64 Intel(R) Core(TM) Ultra 9 185H
    Memory: 37.29 GB / 62.23 GB
    Container: Yes
    Shell: 5.9 - /usr/bin/zsh
  Binaries:
    Node: 22.16.0 - /home/martin/.nvm/versions/node/v22.16.0/bin/node
    npm: 10.9.2 - /home/martin/.nvm/versions/node/v22.16.0/bin/npm
    pnpm: 10.17.1 - /home/martin/.nvm/versions/node/v22.16.0/bin/pnpm
```

### ✒️ Additional context

I'll refactor my code for now, not a big problem.

**Comment by marcalexiei** [comment]
> Unrelated one, but to avoid opening a different issue:

Since you mentioned this is unrelated, it would be best to open a separate issue. 
Reporting multiple unrelated problems in a single issue can cause confusion and make it harder to track and resolve them.

---

> ```ts
>  const schema = z.object({
>   code: z.string(),
>   state: z.string(),
> })
> ```
> Should this one just be fine? 🤷🏼 it's a one-off schema. I'll rename it in my case

Personally I don't like that naming variables just `schema`. 
I think about adding an additional rule setting to allow user pass a name allowlist.
